elephant/phase_analysis.py

```diff
@@ -235,27 +244,35 @@ def pairwise_phase_consistency(phases):
         if phase_array.ndim != 1:
             raise ValueError("Phase arrays must be 1D (use .flatten())")
 
-    result_ppc = []
-
-    for phase_array in phases:
-        n_trials = phase_array.shape[0]
-
-        # Compute the distance between each pair of phases using dot product
-        # Optimize computation time using array multiplications instead of for
-        # loops
-        p_cos_2d = np.tile(np.cos(phase_array), reps=(n_trials, 1))
-        p_sin_2d = np.tile(np.sin(phase_array), reps=(n_trials, 1))
-
-        # By doing the element-wise multiplication of this matrix with its
-        # transpose, we get the distance between phases for all possible pairs
-        # of elements in 'phase'
-        dot_prod = np.multiply(p_cos_2d, p_cos_2d.T) + \
-            np.multiply(p_sin_2d, p_sin_2d.T)
-
-        # Now average over all elements in temp_results (the diagonal are 1
-        # and should not be included)
-        np.fill_diagonal(dot_prod, 0)
-        ppc = 2 / (n_trials * n_trials - n_trials) * np.sum(dot_prod)
-        result_ppc.append(ppc)
-
-    return result_ppc
+    if method not in ['ppc0']:
+        raise ValueError('For method choose out of: ["ppc0"]')
+
+    phase_array = np.hstack(phases)
```

Comment:
are you sure that all phases have the same length?

Reply:
I'm sure all phases _do not_ have the same length :p. I implemented this method (slightly against your earlier advice) after I discovered that simply putting in a whole spiketrain for a 2h session takes too much RAM. As I had to cut up my spiketrain anyway, it felt like it made sense to do so in trials. As such the default input for the function will be (if you agree) a list of arrays, where each array is a trial of size `n_phases`. For the default version of PPC we can simply append all these phases, and compute the consistency, and I'll leave it at this for this PR.

Then in a future PR we can easily extend this function with PPC1, by selecting indices of the input phases, such that we only average between pairs of phases between trials (i.e. PPC1).